fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -153,10 +154,19 @@ def fit(self, X, y, **kwargs):
         )
 
         theta = pd.Series(0, lagrangian.constraints.index)
+
+        # Stores the weights of the best-response estimators without LP and before averaging
         Qsum = pd.Series(dtype="float64")
-        gaps_EG = []
-        gaps = []
-        Qs = []
+
+        # Stores the duality gap of the Exponentiated Gradient (EG) at each iteration
+        gaps_EG: list[float] = []
+
+        # Stores the best gap at each iteration among the ones EG and Linear Programming (LP) yield
+        gaps: list[float] = []
+
+        # Stores the best-reponse estimators' weights at each iteration
+        # The appended weights come either from (EG) or (LP), depending on their respective gaps
+        Qs: list[pd.Series] = []
```

Comment:
I very rarely use type hints outside of function signatures, but when initialization containers for a complex algorithm I think it's helpful for the reader to know what each variable will look like